exe/facter.cc

```diff
@@ -251,7 +255,15 @@ int main(int argc, char **argv)
         facts.add_default_facts(ruby);
 
         if (!vm.count("no-external-facts")) {
+          string inside_facter;
+          environment::get("INSIDE_FACTER", inside_facter);
+
+          if (inside_facter == "true") {
+            LOG_WARNING("Facter was called recursively, skipping external facts. Add '--no-external-facts' to silence this warning");
```

Comment:
It might be worth mentioning that we detected this because INSIDE_FACTER was set to true, in case (somehow) someone accidentally has that in their environment.